token/services/db/sql/init.go

```diff
@@ -0,0 +1,88 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package sql
+
+import (
+	"crypto/sha256"
+	"database/sql"
+	"encoding/hex"
+	"fmt"
+	"regexp"
+
+	"github.com/hyperledger-labs/fabric-smart-client/platform/view/services/flogging"
+	"github.com/pkg/errors"
+)
+
+var logger = flogging.MustGetLogger("token-sdk.sql")
+
+func initSchema(db *sql.DB, schemas ...string) error {
+	logger.Info("creating tables")
+	tx, err := db.Begin()
+	if err != nil {
+		return err
+	}
+	defer tx.Rollback()
+
+	for _, schema := range schemas {
+		logger.Debug(schema)
+		if _, err := db.Exec(schema); err != nil {
+			return errors.Wrap(err, "error creating schema")
+		}
+	}
+	if err = tx.Commit(); err != nil {
+		return err
+	}
+	return nil
+}
+
+type tableNames struct {
+	Movements             string
+	Transactions          string
+	Requests              string
+	Validations           string
+	TransactionEndorseAck string
+	Certifications        string
+	Tokens                string
+	Ownership             string
+	AuditTokens           string
+	IssuedTokens          string
+	PublicParams          string
+	Ledger                string
+}
+
+func getTableNames(prefix, name string) (tableNames, error) {
+	if prefix != "" {
+		r := regexp.MustCompile("^[a-zA-Z_]+$")
```

Comment:
can be a static var

Reply:
isn't this memory that we waste though? We invoke getTableNames only few times.